shared/js/media/media_frame.js

```diff
@@ -144,6 +146,10 @@ MediaFrame.prototype.displayImage = function displayImage(blob,
   this.fullsizeWidth = this.fullSampleSize.scale(width);
   this.fullsizeHeight = this.fullSampleSize.scale(height);
 
+  if (navigator.mozL10n.readyState === 'complete') {
```

Comment:
You could add a comment here to say that if readyState is not 'complete', then localization will have to be handled externally by the same handler that handles language changes.